### ISSUE TITLE
Script errors when run through pq, but not in client

### ISSUE THREAD

**Issue by mclark4386** [cockroachdb/cockroach#21032]
I asked about this in Gitter and was asked to post about it here.

**TL;DR:** I have a [script](https://gist.github.com/mclark4386/c634e75f47f8c362601b9ed9d5e66310) that works if run from `cockroach sql` but fails with `: pq: column "description" does not exist` from my app. However, the [script](https://gist.github.com/mclark4386/e4445d8064a52899bbbcf66b3bb4b8ac) without the last four lines works fine.

**Full Details:**
I am implementing the integration for cockroachDB into a Go database library called [pop](https://github.com/markbates/pop).  When implementing the "change_column" migration feature I figured out that cockroachdb doesn't support in-place column alters that would change the data so I implemented it as:

1. Find the properties of col.
2. Rename col.
3. Create new_col with old name and modified properties.
4. Move all data from renamed_col to new_col.
5. Drop renamed_col.

The script is one of the generated changes. In the app the migration is run inside of a transaction. It almost seems like a race condition to me... like it's trying to run the ALTER before the ADD. If you would like to experiment with this you'll need a "books" table... doesn't really matter what's in it,but in our tests this is how we create it(with a different migration^^):

```
CREATE TABLE "books" (
"id" SERIAL PRIMARY KEY,
"title" VARCHAR (255) NOT NULL,
"user_id" UUID NOT NULL,
"isbn" VARCHAR (50) NOT NULL,
"created_at" timestamp NOT NULL,
"updated_at" timestamp NOT NULL
);
```

It seems important to have the table created before the script is run, but I could be wrong. 

**Things I've tried:**

- Running it from the client (i.e. `cockroach sql`) that worked which was what weirded me out.
- Using an alias for the table in the UPDATE, this caused a different error `: pq: unimplemented: compound types not supported yet: "t.description"`
- surrounding the UPDATE in an EXEC, this caused the error `: pq: syntax error at or near "exec"` (`EXEC('UPDATE "books" SET "description" = "_description_tmp";');`)
- Adding a transaction around just this script, but since it's in a transaction already of course that didn't work
- Few more things I can't remember at the moment

Any help would be GREATLY appreciated! I suspect that I'm either doing something stupid or there is a race going on and I just don't know how to trick it into doing it in the right order...

**Comment by mclark4386** [comment]
So this is now the only bug holding my implementation back. Looking forward to any input y'all can provide!^^ https://github.com/mclark4386/pop/tree/cockroach for those interested.

**Comment by bdarnell** [comment]
Is your app trying to run this script as a single transaction (`cockroach sql` doesn't add transactions on its own; you have to add begin/commit statements yourself)? There are limitations on the use of schema changes inside transactions. In many cases the schema change doesn't take effect until the transaction completes, so you can't do multiple changes in a transaction, or make a schema change and then an update that depends on it. A few combinations are allowed, but in general it's best to either run your schema change statements outside of transactions or in a transaction that only contains one statement.

**Comment by mclark4386** [comment]
It is inside of a single transaction... I'll see what I can do about either breaking the transaction up or removing it completely and report back. Thanks!

**Comment by mclark4386** [comment]
made it so that every change is in it's own transaction. It's ugly as sin, but works perfectly! thanks!

**Comment by vivekmenezes** [comment]
I'd appreciate it if you can list the statements you were originally trying to run in a single transaction. Thanks!

**Comment by mclark4386** [comment]
it's in the gist in the first link from the OP, but to save you the scrolling: https://gist.github.com/mclark4386/c634e75f47f8c362601b9ed9d5e66310
